### PR TITLE
Fix initial resource display status

### LINKED ISSUE #2557: Dashboard resources page displays green tick for empty status
Resources are initially provided to the dashboard with no status. That very quickly changes to `Starting`. However, if you have 100s of resources, it takes a while to start them all. Many wait with no status but have a green tick.

Either no icon should be displayed if the status is empty, or a grey circle (like what `Starting` has)

### PR DESCRIPTION
**Pull request by MatsM16**
Fixes #2557  
When no status:
- Show no status icon
- Show text `Pending`
resources_pending

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2727)

### CODE REVIEW

**Comment by JamesNK** [comment]
@davidfowl What should happen if the resource server returns no status? Options:

* Empty status column
* Or the UI can show a value like "Pending"

**Comment by davidfowl** [comment]
It should say Unknown and be gray. I'm hoping to get the protocol change in that lets the resource server control both the text and icon type that gets shown

File: src/Aspire.Dashboard/Components/ResourcesGridColumns/StateColumnDisplay.razor [hunk 1]
```diff
@@ -30,14 +30,20 @@ else if (Resource is { State: ResourceStates.StartingState })
...
+}
 else
 {
     <FluentIcon Icon="Icons.Filled.Size16.CheckmarkCircle"
                 Color="Color.Success"
                 Class="severity-icon"/>
 }
 
-@Resource.State.Humanize()
+@(Resource.State?.Humanize() ?? "Pending")
```
Comment (JamesNK) [inline src/Aspire.Dashboard/Components/ResourcesGridColumns/StateColumnDisplay.razor:46]:
```suggestion
@(Resource.State?.Humanize() ?? "Unknown")
```

File: src/Aspire.Dashboard/Components/ResourcesGridColumns/StateColumnDisplay.razor [hunk 2]
```diff
@@ -30,14 +30,20 @@ else if (Resource is { State: ResourceStates.StartingState })
                 Color="Color.Info"
                 Class="severity-icon"/>
 }
+else if (Resource is { State: /* pending */ null })
+{
+    /* Show nothing while waiting for
```
Comment (JamesNK) [inline src/Aspire.Dashboard/Components/ResourcesGridColumns/StateColumnDisplay.razor:35]: Use the starting state icon (gray circle)

**Comment by MatsM16** [comment]
## Now looks like this

Screenshot 2024-03-08 094416

**Review by JamesNK — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**